tidy3d/plugins/adjoint/components/simulation.py

```diff
@@ -227,18 +225,25 @@ def _warn_if_colocate(cls, val):
         return val
 
     @staticmethod
-    def get_freq_adjoint(output_monitors: List[Monitor]) -> float:
-        """Return the single adjoint frequency stripped from the output monitors."""
+    def get_freqs_adjoint(output_monitors: List[Monitor]) -> List[float]:
+        """Return sorted list of unique frequencies stripped from a collection of monitors."""
 
         if len(output_monitors) == 0:
             raise AdjointError("Can't get adjoint frequency as no output monitors present.")
 
-        return output_monitors[0].freqs[0]
+        freqs = []
+        for mnt in output_monitors:
+            for freq in mnt.freqs:
+                if freq not in freqs:
+                    freqs.append(freq)
+
+        freqs.sort()
+        return freqs
```

Comment:
probably fewer lines of code by using `numpy.unique`?

Reply:
thanks, made this change.